main.go

```diff
@@ -66,53 +72,114 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
-	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
-	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
-	metrics.Registry.MustRegister(common.SSPOperatorReconcileSucceeded)
-	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
-	mux := http.NewServeMux()
-	mux.Handle("/metrics", handler)
+// This callback executes on each client call returning a new config to be used
+// please be aware that the APIServer is using http keepalive so this is going to
+// be executed only after a while for fresh connections and not on existing ones
+func getConfigForClientCallback(cfg *tls.Config, apiClient client.Client, ctx context.Context) (*tls.Config, error) {
+	var err error
+	var tlsProfile *ocpconfigv1.TLSSecurityProfile
 
-	minTlsVersion, err := tlsOptions.MinTLSVersionId()
-	if err != nil {
-		return err
+	var sspList ssp.SSPList
```

Comment:
Why did you revert back to an explicit API call per callback? Please keep a cached version / a shared informer of the SSP CR.

Reply:
This is the client that is passed from manager and it should be using cache.

Reply:
It was my idea. The `apiClient` uses a cache internally. It creates a shared informer for any requested resource. So calling `Get` and `List` methods on the client does not make an API call.

Reply:
ACK